sklearn/linear_model/tests/test_sgd.py

```diff
@@ -113,1102 +140,1181 @@ def decision_function(self, X, *args, **kw):
-
-    def test_partial_fit_equal_fit_optimal(self):
-        self._test_partial_fit_equal_fit("optimal")
-
-    def test_partial_fit_equal_fit_invscaling(self):
-        self._test_partial_fit_equal_fit("invscaling")
-
-    def test_partial_fit_equal_fit_adaptive(self):
-        self._test_partial_fit_equal_fit("adaptive")
-
-    def test_regression_losses(self):
-        clf = self.factory(alpha=0.01, learning_rate="constant",
-                           eta0=0.1, loss="epsilon_insensitive")
+    for loss in ("hinge", "squared_hinge", "log", "modified_huber"):
+        clf = klass(penalty='l2', alpha=0.01, fit_intercept=True,
+                    loss=loss, max_iter=10, shuffle=True)
         clf.fit(X, Y)
-        assert_equal(1.0, np.mean(clf.predict(X) == Y))
+        # assert_almost_equal(clf.coef_[0], clf.coef_[1], decimal=7)
+        assert_array_equal(clf.predict(T), true_result)
 
-        clf = self.factory(alpha=0.01, learning_rate="constant",
-                           eta0=0.1, loss="squared_epsilon_insensitive")
-        clf.fit(X, Y)
-        assert_equal(1.0, np.mean(clf.predict(X) == Y))
 
-        clf = self.factory(alpha=0.01, loss="huber")
-        clf.fit(X, Y)
-        assert_equal(1.0, np.mean(clf.predict(X) == Y))
+@pytest.mark.parametrize('klass', [SGDClassifier, SparseSGDClassifier])
+def test_sgd_bad_l1_ratio(klass):
+    # Check whether expected ValueError on bad l1_ratio
+    assert_raises(ValueError, klass, l1_ratio=1.1)
 
-        clf = self.factory(alpha=0.01, learning_rate="constant", eta0=0.01,
-                           loss="squared_loss")
-        clf.fit(X, Y)
-        assert_equal(1.0, np.mean(clf.predict(X) == Y))
 
-    def test_warm_start_multiclass(self):
-        self._test_warm_start(X2, Y2, "optimal")
+@pytest.mark.parametrize('klass', [SGDClassifier, SparseSGDClassifier])
+def test_sgd_bad_learning_rate_schedule(klass):
+    # Check whether expected ValueError on bad learning_rate
+    assert_raises(ValueError, klass, learning_rate="<unknown>")
 
-    def test_multiple_fit(self):
-        # Test multiple calls of fit w/ different shaped inputs.
-        clf = self.factory(alpha=0.01, shuffle=False)
-        clf.fit(X, Y)
-        assert hasattr(clf, "coef_")
 
-        # Non-regression test: try fitting with a different label set.
-        y = [["ham", "spam"][i] for i in LabelEncoder().fit_transform(Y)]
-        clf.fit(X[:, :-1], y)
+@pytest.mark.parametrize('klass', [SGDClassifier, SparseSGDClassifier])
+def test_sgd_bad_eta0(klass):
+    # Check whether expected ValueError on bad eta0
+    assert_raises(ValueError, klass, eta0=0,
+                  learning_rate="constant")
 
 
-class SparseSGDClassifierTestCase(DenseSGDClassifierTestCase):
-    """Run exactly the same tests using the sparse representation variant"""
+@pytest.mark.parametrize('klass', [SGDClassifier, SparseSGDClassifier,
+                                   SGDRegressor, SparseSGDRegressor])
+def test_sgd_bad_alpha(klass):
```

Comment:
this is where you included the regressors in the tests compared to master, which is fine I guess.